src/components/actions/extendApplicationRoute.ts

```diff
@@ -0,0 +1,63 @@
+import { combineRequestParams } from '../utilityFunctions'
+import { PermissionPolicyType, Trigger } from '../../generated/graphql'
+import { DateTime } from 'luxon'
+import DBConnect from '../databaseConnect'
+import { getPermissionNamesFromJWT } from '../data_display/helpers'
+
+export const routeExtendApplication = async (request: any, reply: any) => {
+  const { applicationId, eventCode, extensionTime, data } = combineRequestParams(request, 'camel')
+
+  // Check permissions first -- currently we just check user has ANY Reviewing
+  // permissions for current template.
+  // TO-DO: define specific permission names required for extending deadlines
+  const { permissionNames: userPermissions } = await getPermissionNamesFromJWT(request)
+
+  const templatePermissions = (
+    await DBConnect.getTemplatePermissionsFromApplication(Number(applicationId))
```

Comment:
Coercing `applicationId` to Number here, as it could come in on query parameters (no just body JSON), in which case it is always a string. The node-postgres module doesn't care, but GraphQL very much does.